connexion/lifecycle.py

```diff
@@ -34,4 +34,21 @@ def __init__(self,
         self.mimetype = mimetype
         self.content_type = content_type
         self.body = body
-        self.headers = headers or {}
+        self.headers = ConnexionHeaders(headers)
+
+
+class ConnexionHeaders(object):
```

Comment:
I would vote for migrating the representation to the list of tuples in case someone pass a dict to the constructor of `ConnexionHeaders`. This is an internal API for Connexion. But I would implement here a method `get(name, default=None)`.